src/ucp/core/ucp_mm.h

```diff
@@ -181,6 +181,7 @@ static UCS_F_ALWAYS_INLINE size_t ucp_memh_length(const ucp_mem_h memh)
 #define UCP_MEM_IS_ACCESSIBLE_FROM_CPU(_mem_type) \
     (UCS_BIT(_mem_type) & UCS_MEMORY_TYPES_CPU_ACCESSIBLE)
 #define UCP_MEM_IS_GPU(_mem_type) ((_mem_type) == UCS_MEMORY_TYPE_CUDA || \
+                                   (_mem_type) == UCS_MEMORY_TYPE_CUDA_MANAGED || \
```

Comment:
so we want to treat managed memory same as device memory?

Reply:
@yosefe Found out in different thread because UCX treats managed memory as host, it results in compute time being affected as UCX communication phases (through shmem for example) end up touching managed memory regions which prompts page migration of device resident pages to host memory, and subsequently leads to slower compute time (as the pages have to migrate back from host if compute is re-initiated on the same memory region on the GPU -- which is a common case). In light of this, we do want managed memory treated exactly like pinned memory and copy user memory instead of causing page migrations.